beetsplug/importfeeds.py

```diff
@@ -84,20 +84,38 @@ def __init__(self):
         else:
             self.config["relative_to"] = self.get_feeds_dir()
 
+        self.m3u_session = None
+
         self.register_listener("album_imported", self.album_imported)
         self.register_listener("item_imported", self.item_imported)
         self.register_listener("import_begin", self.import_begin)
 
+    @property
+    def formats(self):
+        """Get formats list, handling confuse exceptions gracefully."""
+        try:
+            return self.config["formats"].as_str_seq()
+        except Exception:
+            return []
+
     def get_feeds_dir(self):
         feeds_dir = self.config["dir"].get()
         if feeds_dir:
             return os.path.expanduser(bytestring_path(feeds_dir))
-        return config["directory"].as_filename()
+
+        # Fallback to global config directory, with error handling
+        try:
+            return config["directory"].as_filename()
+        except Exception:
+            # If global config not available (e.g., in tests), use temp dir
+            import tempfile
+
+            return tempfile.gettempdir()
```

Comment:
I added this to fix tests `test_importhistory` and `test_importfeeds` failing when running alongside each other (eg. in CI)

I'm really not sure if this only fixes the symptoms instead of a proper fix.... Another eye please @semohr @snejus Thanks in advance!

Reply:
Reverted my changes, there you go: https://github.com/beetbox/beets/actions/runs/18250250353/job/51963721156?pr=4748

Thanks for looking into it!

Reply:
Apologies - I haven't. Will check this now